packages/react-devtools-shared/src/backend/fiber/renderer.js

```diff
@@ -2128,6 +2133,54 @@ export function attach(
     pendingStringTableLength = 0;
   }
 
+  function measureHostInstance(instance: HostInstance): null | Array<Rect> {
+    // Feature detect measurement capabilities of this environment.
+    // TODO: Consider making this capability injected by the ReactRenderer.
+    if (typeof instance !== 'object' || instance === null) {
+      return null;
+    }
+    if (typeof instance.getClientRects === 'function') {
+      // DOM
+      return Array.from(instance.getClientRects());
+    }
+    if (instance.canonical) {
+      // Native
+      const publicInstance = instance.canonical.publicInstance;
+      if (!publicInstance) {
+        // The publicInstance may not have been initialized yet if there was no ref on this node.
+        // We can't initialize it from any existing Hook but we could fallback to this async form:
+        // renderer.extraDevToolsConfig.getInspectorDataForInstance(instance).hierarchy[last].getInspectorData().measure(callback)
```

Comment:
@hoxyq The current technique is required to be async which should be fine with the new React Native APIs. However, currently we can only read the size this way if the publicInstance has been initialized which is now lazy. https://github.com/facebook/react/pull/33943

So this works sometimes but not always.

We don't have access to initialize this from the injected hooks. We do have access to the `extraDevToolsConfig` which has some info we can use. However, that's only available as an async API which would require a more complex implementation here. That would work for old versions too.

We could also add another injected devtools hook for measuring instances but that would require upgrading React Native versions to get them.